### ISSUE TITLE
react/display-name false positive when using Object.defineProperty

### ISSUE THREAD

**Issue by goldhand** [yannickcr/eslint-plugin-react#1780]
I'm using `Object.defineProperties` to define a component's `displayName` but eslint is ignoring this and giving me a `react/display-name` error. I'm not sure if this is an `eslint-plugin-react` or downstream issue.

Example:
```js
const Foo = props => <p>foo</p>;
Object.defineProperty(Foo, 'displayName', {
  value: 'Foo',
});
```
will give me a `react/display-name` error.

as will defining additional parameters such as `enumerable`.

env:
```
node: >=8

    "babel-eslint": "^7.2",
    "eslint": "^4.10",
    "eslint-plugin-babel": "^4.1",
    "eslint-plugin-flowtype": "^2.33",
    "eslint-plugin-import": "^2.7",
    "eslint-plugin-jsx-a11y": "^5.1",
    "eslint-plugin-react": "^7.4"

```

**Comment by ljharb** [comment]
The linter rule can't track all the possible ways you could mutate a component to add a display name; you could use `_.extend(Foo, { displayName: 'foo' })` as well, and there's no way to statically know what it's doing.

One point of this rule is that you shouldn't use an arrow function as an SFC - you *must* use a normal named function.
